src/instana/agent/host.py

```diff
@@ -346,14 +347,17 @@ def report_spans(self, payload: Dict[str, Any]) -> Optional[Response]:
         return
 
     def filter_spans(self, spans: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
-        from instana.util.traceutils import is_service_or_endpoint_ignored
-
+        """
+        Before propagating spans to the agent, this function is being called to filter
+        span list using ignore-endpoint variable.
+        """
```

Comment:
Ok, this is not clear to me. What is this function really doing? 🤷🏻‍♂️
Is the return a list of spans that I should ignore or a list of valid spans to send to the agent?